keras_nlp/metrics/rouge_l_test.py

```diff
@@ -0,0 +1,248 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for RougeL."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.metrics import RougeL
+
+
+class RougeLTest(tf.test.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.metric_types = (
+            "rouge-l_precision",
+            "rouge-l_recall",
+            "rouge-l_f1_score",
+        )
+
+    def test_initialization(self):
+        rouge = RougeL()
+        result = rouge.result()
+
+        for metric_type in self.metric_types:
+            self.assertEqual(result[metric_type].numpy(), 0.0)
+
+    def test_string_input(self):
+        rouge = RougeL(use_stemmer=False)
+        y_true = "the tiny little cat was found under the big funny bed"
+        y_pred = "the cat was under the bed"
+
+        rouge_val = rouge(y_true, y_pred)
+        for metric_type, expected_val in zip(
+            self.metric_types, [1, 0.545, 0.706]
+        ):
+            self.assertAlmostEqual(
+                rouge_val[metric_type].numpy(), expected_val, delta=1e-3
+            )
+
+    def test_string_list_input(self):
+        rouge = RougeL(use_stemmer=False)
+        y_true = [
+            "the tiny little cat was found under the big funny bed",
+            "i really love contributing to KerasNLP",
+        ]
+        y_pred = [
+            "the cat was under the bed",
+            "i love contributing to KerasNLP",
+        ]
+
+        rouge_val = rouge(y_true, y_pred)
+        for metric_type, expected_val in zip(
+            self.metric_types, [1, 0.689, 0.807]
+        ):
+            self.assertAlmostEqual(
```

Comment:
Can we assert a whole dict structure in here? If so I would fine that a lot more readable actually than the way it's done here. Here and elsewhere

```
assertAlmostEqual(rouge_output, {
    "rouge-l_precision": x,
    "rouge-l_recall": y,
    "rouge-l_f1_score": z,
})
```

Reply:
Right - I can make a custom function for this!